### PR TITLE
Machine-code cleanup

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

part 2 of machine part stacks.
Fully removes machinepart prototypes and just uses stacks and components for construction.
removes all the associated code around machine parts
cleans up code and adds localization for machine construction that previously didn't have it.
adds protoId validation and a test for validating `ComponentRequirements`

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

MachinePartPrototype and MachinePartComponent are now fully removed.
Machine boards using `Requirements` have been converted `MaterialRequirements` and `ComponentRequirements`.
PartExchanger and upgrade code have been fully removed.

**Changelog**

no cl no fun

### CODE REVIEW

**Comment by mirrorcult** [comment]
-600 diff Damn!!

**Comment by mirrorcult** [comment]
wrong. button

**Comment by EmoGarbage404** [comment]
> wrong. button

1984

File: Content.Shared/Construction/Components/MachineBoardComponent.cs [hunk 1]
```diff
@@ -1,54 +1,50 @@
...
+public sealed partial class MachineBoardComponent : Component
+{
+    /// <summary>
+    /// The stacks needed to construct this machine
+    /// </summary>
+    /// <remarks>
+    /// MISNAMED VARIABLE: this refers to stacks, not materials.
+    /// </remarks>
+    [DataField]
+    public Dictionary<ProtoId<StackPrototype>, int> MaterialRequirements = new();
```
Comment (AJCM-git) [inline Content.Shared/Construction/Components/MachineBoardComponent.cs:18]: it takes like a F2 button press and a regex so do it

**Review by AJCM-git — APPROVED** [review]
just do this and self merge, also, does this fix the bug where tag requirements dont tick down in the examine when you add them?

**Pull request merged**